soroban-auth/tests/test.rs

```diff
@@ -14,31 +13,34 @@ pub enum DataKey {
     Nonce(Identifier),
 }
 
-fn read_nonce(e: &Env, id: Identifier) -> BigInt {
-    let key = DataKey::Nonce(id);
+fn read_nonce(e: &Env, id: &Identifier) -> BigInt {
+    let key = DataKey::Nonce(id.clone());
     e.contract_data()
         .get(key)
         .unwrap_or_else(|| Ok(BigInt::zero(e)))
         .unwrap()
 }
 
-struct NonceForSignature(Signature);
-
-impl NonceAuth for NonceForSignature {
-    fn read_nonce(e: &Env, id: Identifier) -> BigInt {
-        read_nonce(e, id)
+fn verify_and_consume_nonce(e: &Env, id: &Identifier, expected_nonce: &BigInt) {
+    // replay protection is not required for Contract authorization because
+    // there's no cryptographic signature involved. All that's checked is the
+    // invoking contract, so this contract just expects 0.
+    match id {
+        Identifier::Contract(_) => {
+            if BigInt::zero(&e) != *expected_nonce {
+                panic!("nonce should be zero for Contract")
+            }
+        }
+        _ => {}
     }
 
-    fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt {
-        let key = DataKey::Nonce(id.clone());
-        let nonce = Self::read_nonce(e, id);
-        e.contract_data().set(key, &nonce + 1);
-        nonce
-    }
+    let key = DataKey::Nonce(id.clone());
+    let nonce = read_nonce(e, id);
 
-    fn signature(&self) -> &Signature {
-        &self.0
+    if nonce != *expected_nonce {
```

Comment:
Does this equality check not work without the `*`? I think I fixed that recently, it should work now.

Reply:
No I get this error
```
can't compare `soroban_sdk::BigInt` with `&soroban_sdk::BigInt`
the trait `PartialEq<&soroban_sdk::BigInt>` is not implemented for `soroban_sdk::BigInt`
```

Reply:
Ok no worries. I'll fix it, but no need to do anything differently than you are here.

Reply:
Fixing in https://github.com/stellar/rs-soroban-sdk/pull/598.

Reply:
updated